### ISSUE TITLE
Can we remove `ParallelRunner`?

### ISSUE THREAD

**Issue by merelcht** [kedro-org/kedro#4291]
## Description
Check telemetry to find out how much `ParallelRunner` is used and whether it would be possible to deprecate and remove it.

## Context
`ParellelRunner` is the most problematic among our runners:
- https://github.com/kedro-org/kedro/issues/4078
- https://github.com/kedro-org/kedro-viz/issues/1801
- https://github.com/kedro-org/kedro/issues/3713
- https://github.com/kedro-org/kedro/issues/4176
- https://github.com/kedro-org/kedro/issues/2162
- https://github.com/kedro-org/kedro/issues/3804
- https://github.com/kedro-org/kedro/issues/3674
- https://github.com/kedro-org/kedro/issues/673

And there's already a bit of evidence it's not used that much: https://linen-slack.kedro.org/t/16663577/do-you-use-kedro-run-runner-parallerunner-to-speed-up-your-p#99abccb0-7970-4a65-8fad-85fd22681beb

**Comment by astrojuanlu** [comment]
We discussed that there is indeed evidence that some people at least _try_ to use this runner, but because it's so broken, they're rarely successful or they have to battle workarounds.

We agreed to do some research to understand the underlying issues and see if the `ParallelRunner` is still the best solution to them.

**Comment by yury-fedotov** [comment]
I'm using `ParallelRunner` in most projects... Surprised if its unpopular, based on telemetry

**Comment by astrojuanlu** [comment]
Thanks @yury-fedotov, that's good to know. Does it work well for your needs? Have you been hit by any issues?

**Comment by tejaassolanki** [comment]
Even I have used it in my previous organisation and it is most likely still being used. If I remember correctly, the telemetry data won't show up from that organisation.

**Comment by astrojuanlu** [comment]
Unsure if this is being recorded in the telemetry at all. The prompt for this issue was a Slack thread that isn't necessarily representative.

I'm mostly ready to say that we should fix the issues instead of deprecating the runner.

**Comment by astrojuanlu** [comment]
Another user question with the `ParallelRunner` https://kedro.hall.community/parallel-runner-memory-utilization-issue-with-modular-pipelines-DcHp4Zt3SPx2

I think it would be too disruptive to remove the `ParallelRunner`, we should instead try to fix its issues (which are many). Do you agree @merelcht ?

**Comment by merelcht** [comment]
@astrojuanlu yes agreed! It definitely looks like `ParallelRunner` is used by people.
